lib/sourcemaps.js

```diff
@@ -71,27 +73,28 @@ exports.concatenateSourceMaps = function(sourceFilename, mapsWithOffsets, source
 
   // convert from library internals format to canonical
   var normalized = JSON.parse(JSON.stringify(generated));
+  var sourcePath;
 
   if (includeSourcesContent) {
     normalized.sourcesContent = normalized.sources.map(function(source) {
     //if (contentsBySource[source])
       //return contentsBySource[source];
 
-    if (source.substr(0, 6) === 'file:/')
-      return fs.readFileSync(source.replace(/^file:\/+/, '/')).toString();
+    sourcePath = filePath(source);
+    if (sourcePath)
+      return fs.readFileSync(sourcePath).toString();
 
     // remove this is optimistic return is reliable
     return contentsBySource[source];
     });
   }
 
-  // convert paths to relative
+  var rootIsPath = isFileURL(sourceRoot);
```

Comment:
The intention here is to support a non-file URL, rather than a path. Can drop the non-used flexibility.